File: domusviewer/app/src/main/java/it/achdjian/paolo/domusviewer/DomusEngineRest/RequestIdentify.java
package it.achdjian.paolo.domusviewer.DomusEngineRest;

import android.content.SharedPreferences;
import android.support.annotation.NonNull;

import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.IOException;

/**
 * Created by Paolo Achdjian on 15/04/16.
 */
public class RequestIdentify extends DomusEngineRest {
    private final int nwkAddress;
    private final int endpoint;

    public RequestIdentify(@NonNull SharedPreferences sharedPreferences, @NonNull ConnectionStatus connected, int nwkAddress, int endpoint) {
        super(sharedPreferences, connected);
        this.nwkAddress = nwkAddress;
        this.endpoint = endpoint;
    }

    @Override
    public void run() {
        post("/devices/" + Integer.toString(nwkAddress, 16) + "/endpoint/" + Integer.toString(endpoint, 16) + "/cluster/in/3/command/0");
    }
}

File: ZTopology/app/src/main/java/it/achdjian/paolo/ztopology/rest/RequestIdentify.kt
package it.achdjian.paolo.ztopology.domusEngine.rest

import android.util.Log
import it.achdjian.paolo.ztopology.domusEngine.rest.ZigbeeRunnable
import it.achdjian.paolo.ztopology.domusEngine.rest.DomusEngineRest

/**
 * Created by Paolo Achdjian on 8/23/17.
 */
class RequestIdentify (val networkdId: Int, val endpointId: Int): ZigbeeRunnable() {
    override fun run() {
        val builder = StringBuilder("/devices/").
                append(Integer.toString(networkdId, 16)).
                append("/endpoint/").
                append(Integer.toString(endpointId, 16)).
                append("/cluster/in/3/command/0")
        Log.d(TAG, builder.toString())
        DomusEngineRest.post(builder.toString())
    }
}